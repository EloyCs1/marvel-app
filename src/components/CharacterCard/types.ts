import { AppCharacter } from "src/service/types";

export interface CharacterCardProps {
  character: AppCharacter.Character;
}
