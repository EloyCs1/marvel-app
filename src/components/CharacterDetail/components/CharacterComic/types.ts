import { AppCharacter } from "src/service/types";

export interface CharacterComicProps {
  comic: AppCharacter.Comic;
}
