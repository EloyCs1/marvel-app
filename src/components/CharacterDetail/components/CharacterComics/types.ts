import { AppCharacter } from "src/service/types";

export interface CharacterComicsProps {
  comics: AppCharacter.Comic[];
  loading: boolean;
}
