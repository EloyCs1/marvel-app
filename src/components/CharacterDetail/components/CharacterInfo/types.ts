import { AppCharacter } from "src/service/types";

export interface CharacterInfoProps {
  detail: AppCharacter.Character;
  loading: boolean;
}
