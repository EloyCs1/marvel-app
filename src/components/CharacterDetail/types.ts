export interface CharacterDetailProps {
  characterId: number;
}
