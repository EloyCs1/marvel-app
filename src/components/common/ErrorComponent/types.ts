export interface ErrorComponentProps {
  testId?: string;
  tryAgain?: () => void;
}
