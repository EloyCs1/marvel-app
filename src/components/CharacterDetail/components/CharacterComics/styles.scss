.characterDetail {
  &__comics {
    padding: 48px 48px 0 48px;
    display: grid;
    gap: 24px;

    &-title {
      text-transform: uppercase;
      margin: auto;
      width: 960px;
      font-size: 32px;
      font-weight: 700;
      line-height: normal;
    }

    &-section {
      width: 960px;
      display: flex;
      gap: 16px;
      overflow-x: auto;
      margin: auto;
      min-height: 229px;

      .loader {
        width: 100%;
        align-content: center;
      }
    }

    @media (max-width: 1008px) {
      padding: 48px 0 0 48px;
      &-title {
        width: 100%;
      }

      &-section {
        width: 100%;
      }
    }

    @media (max-width: 575px) {
      padding: 48px 0 0 16px;
    }
  }
}
