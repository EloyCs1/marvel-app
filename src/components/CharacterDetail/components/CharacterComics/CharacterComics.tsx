import { useIntl } from "react-intl";

import Loader from "src/components/common/Loader/Loader";
import CharacterComic from "../CharacterComic/CharacterComic";

import { CharacterComicsProps } from "./types";
import "./styles.scss";

const CharacterComics: React.FC<CharacterComicsProps> = ({ comics, loading }) => {
  const { formatMessage } = useIntl();

  return (
    <div className="characterDetail__comics">
      <p className="characterDetail__comics-title">{formatMessage({ id: "characterDetail.comics.title" })}</p>
      <div className="characterDetail__comics-section">
        {loading ? <Loader /> : comics.map((comic) => <CharacterComic key={comic.id} comic={comic} />)}
      </div>
    </div>
  );
};
export default CharacterComics;
