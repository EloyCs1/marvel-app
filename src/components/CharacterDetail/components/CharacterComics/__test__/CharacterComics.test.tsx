import { render } from "@testing-library/react";
import { COMICS } from "src/mocks/mocks";
import { AppWrapper } from "src/test/test.utils";
import { CharacterComicsProps } from "../types";
import CharacterComics from "../CharacterComics";

jest.mock("src/components/CharacterDetail/components/CharacterComic/CharacterComic", () => ({
  __esModule: true,
  default: () => <div></div>,
}));

const defaultProps: CharacterComicsProps = {
  comics: COMICS,
  loading: false,
};

describe("CharacterComics", () => {
  test("Default render", () => {
    const wrapper = render(
      <AppWrapper>
        <CharacterComics {...defaultProps} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("Loading render", () => {
    const wrapper = render(
      <AppWrapper>
        <CharacterComics {...defaultProps} loading={true} />
      </AppWrapper>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
