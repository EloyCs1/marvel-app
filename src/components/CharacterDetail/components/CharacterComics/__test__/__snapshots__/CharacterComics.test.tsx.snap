// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterComics Default render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="characterDetail__comics"
      >
        <p
          class="characterDetail__comics-title"
        >
          Comics
        </p>
        <div
          class="characterDetail__comics-section"
        >
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="characterDetail__comics"
    >
      <p
        class="characterDetail__comics-title"
      >
        Comics
      </p>
      <div
        class="characterDetail__comics-section"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CharacterComics Loading render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="characterDetail__comics"
      >
        <p
          class="characterDetail__comics-title"
        >
          Comics
        </p>
        <div
          class="characterDetail__comics-section"
        >
          <div
            class="loader"
          >
            <span
              class="loader__spinner"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="characterDetail__comics"
    >
      <p
        class="characterDetail__comics-title"
      >
        Comics
      </p>
      <div
        class="characterDetail__comics-section"
      >
        <div
          class="loader"
        >
          <span
            class="loader__spinner"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
